src/storage-types/src/sources.rs

```diff
@@ -687,6 +687,9 @@ pub trait SourceConnection: Debug + Clone + PartialEq + AlterCompatible {
     /// details of that export, else is set to `SourceExportDetails::None` to indicate that
     /// this source should not export to the primary collection.
     fn primary_export_details(&self) -> SourceExportDetails;
+
+    /// Whether the source type supports read only mode.
+    fn supports_read_only(&self) -> bool;
```

Comment:
I think this is too coarse-grained and we want to also look at the envelope. Maybe something like `SourceEnvelope::wants_rehydration()`? And then at the call-sites we check `supports_read_only() && wants_rehydration()`, wdyt?

Reply:
I put up https://github.com/MaterializeInc/materialize/pull/29798 for discussion.